src/main/java/org/spongepowered/common/world/storage/SpongePlayerDataHandler.java

```diff
@@ -65,48 +64,45 @@
     private final Map<UUID, SpongePlayerData> playerDataMap = new ConcurrentHashMap<>();
 
     public static void init() {
-        SpongePlayerDataHandler handlerInstance = Holder.INSTANCE;
-        if (handlerInstance.hasInitialized) {
+        final SpongePlayerDataHandler handlerInstance = Holder.INSTANCE;
+        if (!Sponge.isServerAvailable() || handlerInstance.hasInitialized) {
             return;
         }
-        @Nullable File root = DimensionManager.getCurrentSaveRootDirectory();
-        if (root == null) {
-            return;
-        }
-        final Path directoryRoot = root.toPath();
-        if (root != null) { // ok, we're on the server, guaranteed.
-            final String filePath = directoryRoot.toString() + File.separator + "data" + File.separator + SPONGE_DATA;
-            try {
-                handlerInstance.playerDir = Paths.get(filePath);
-                Path file = Files.createDirectories(handlerInstance.playerDir);
-                List<Path> playerFiles = new ArrayList<>();
-                try (DirectoryStream<Path> stream = Files.newDirectoryStream(file, "*.{dat}")) {
-                    for (Path entry : stream) {
-                        playerFiles.add(entry);
-                    }
-                } catch (DirectoryIteratorException e) {
-                    SpongeImpl.getLogger().log(Level.ERROR, "Something happened when trying to gather all player files", e);
+
+        final Path filePath = DimensionManager.getCurrentSavesDirectory().get().resolve("data").resolve
+                (SPONGE_DATA);
+
+        try {
+            handlerInstance.playerDir = filePath;
+            Files.createDirectories(handlerInstance.playerDir);
+
+            final List<Path> playerFiles = new ArrayList<>();
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(filePath, "*.{dat}")) {
+                for (Path entry : stream) {
+                    playerFiles.add(entry);
                 }
-                for (Path playerFile : playerFiles) {
-                    File actualFile = playerFile.toFile();
-                    if (actualFile.exists() && actualFile.isFile()) {
-                        FileInputStream stream = new FileInputStream(actualFile);
-                        NBTTagCompound compound = CompressedStreamTools.readCompressed(stream);
-                        stream.close();
-                        DataContainer container = NbtTranslator.getInstance().translateFrom(compound);
-                        SpongePlayerData data = container.getSerializable(DataQuery.of(), SpongePlayerData.class).get();
-                        handlerInstance.playerDataMap.put(data.uuid, data);
-                    }
+            } catch (DirectoryIteratorException e) {
+                SpongeImpl.getLogger().error("Something happened when trying to gather all player files", e);
+            }
+            for (Path playerFile : playerFiles) {
+                File actualFile = playerFile.toFile();
```

Comment:
Is there a reason why you use `java.io.File` here? The methods for `Path` would be `Files.isReadable` or `Files.isRegularFile` ot something like that, and then `Files.newInputStream`